src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -230,11 +230,38 @@ def read_file_content(file_path, allow_binary=False):
     raise CLIError('Failed to decode file {} - unknown decoding'.format(file_path))
 
 
+def preprocess_json(original):
+    lines = original.splitlines()
```

Comment:
What part of this code deals with multi-line breaks?  I.e. stuff like this:

    "multiline1": {
      "type": "string",
      "value": "this string has two
        lines"
    },
    "multiline2": {
      "type": "string",
      "value": "[a('')]  // this is not a comment, it's
        part of the first line
        and this is another line"
    }

Thx!

Do you need some unit tests?

Reply:
yes, it would be great if you could send me some tests!

Regarding multi-line breaks, perhaps I am doing it wrong. Basically I am concatenating all teh lines, but now when i see this example i assume there must be some special algorithm that also takes care of the space.
Is it described somewhere?

Reply:
Unfortunately I don't think so.  Basically their parser (Newtonsoft) doesn't consider \r or \n to end a string and simply collects those characters and continues until it finds the end of string.  

Assuming you need to preprocess before handing it to your parser, it seems like the best way to do this is probably to simply turn \n into the two characters "\" and "n", ditto for \r.  That I think should give you equivalence values, say if you need to grab default parameter values. If you ever give errors with line numbers, you'll need to adjust the line/column numbers appropriately.  Or add the \n\r after the string, so that code after the multi-line string still has the same line number.

Start by looking at the examples here: https://github.com/bmoore-msft/arm-template-language-samples

Reply:
BTW, you are passing the original, unaltered JSON to the backend, right, and the altered one is just used for parameter parsing, correct?  This is important.

Reply:
hmm... that's getting complicated right now.
There are 2 scenarios in Azure CLI right now.

(1) ARM template is specified via URI.
In this case Azure CLI is doing only validation, and later only URI is passed.
So that's pretty simple to handle.

(2) ARM is loaded from the file
In this case Azure CLI is doing validation, but also json.loads / json.dumps.
That is most likely done to replace '' with "" which may sometimes happen.
Perhaps it handles something else as well.
json.loads can't handle comments or multiline strings.

Reply:
For #2, do you currently pass an altered version of the JSON to the backend? You can modify it in order to load the JSON for processing parameters, but is there a reason you can't pass the original file contents as a string to ARM?

cc @bmoore-msft

Reply:
you'll want to pass the original, unaltered template to ARM.  We calculate a templateHash on the template and if you change it the user will never be able to map it back to their source.

Reply:
Yes, CLI can supply the original template to the ARM service. 
The main issue is how many cases that Newton JSON  handles which the default Python JSON parser doesn't? Seems like we need to do a Python version of  Newton JSON? Or the handling 2 cases mentioned in #8937 are sufficient, which are the comment and multi-line strings?